### PR TITLE
REFACTOR-#5416: fix `FutureWarning: the mangle_dupe_cols keyword is deprecated` for `read_excel`

### LINKED ISSUE #5416: CLN: `FutureWarning: the mangle_dupe_cols keyword is deprecated` for `read_excel`
This warning appears for `python` engine.

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Anatoly Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5416 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Review by mvashishtha — APPROVED** [review]

File: modin/core/io/io.py [hunk 1]
```diff
@@ -278,67 +278,9 @@ def read_clipboard(cls, sep=r"\s+", **kwargs):  # pragma: no cover # noqa: PR01
...
-        skip_footer=0,
-        skipfooter=0,
-        convert_float=True,
-        mangle_dupe_cols=True,
-        na_filter=True,
-        **kwds,
-    ):  # noqa: PR01
-        if skip_footer != 0:
-            skipfooter = skip_footer
+    def read_excel(cls, **kwargs):  # noqa: PR01
```
Comment (YarShev) [inline modin/core/io/io.py:281]: Should we clean up this module by replacing keyword arguments of methods exactly to **kwargs as it is done here? Feels some inconsistency between methods. In some places **kwargs, in others exactly keyword args as is.

**Review by YarShev — COMMENTED** [review]
LGTM! Left a question.

File: modin/core/io/io.py (hunk 1 above)
Comment (anmyachev) [reply modin/core/io/io.py:281]: I was planning to try this.

**Review by vnlitvinov — COMMENTED** [review]
While this saves a bunch of lines, this 

a) hinders hinting in an IDE, and
b) limits our ability to override things - that IO function is now assumed to always have parameters which are understood by `pandas.read_excel` (and that assumption could be false).

**Comment by anmyachev** [comment]
> that IO function is now assumed to always have parameters which are understood by `pandas.read_excel` (and that assumption could be false).

Is it bad for the base class? It just passes parameters to pandas.

**Comment by anmyachev** [comment]
> a) hinders hinting in an IDE

@vnlitvinov this is true, but now we almost everywhere have this approach for IO functions and I am ready to put up with it so as not to duplicate the interface code for different engines.

Do you have any suggestions how to solve these two problems?
Сan we merge these changes?

**Comment by dchigarev** [comment]
> Do you have any suggestions how to solve these two problems?

We can probably borrow an idea from pandas' [`deprecate_kwarg` decorator](https://github.com/pandas-dev/pandas/blob/2e218d10984e9919f0296931d92ea851c6a6faf5/pandas/io/parsers/readers.py#L854).

**Comment by anmyachev** [comment]
@mvashishtha let's merge it since there are no change requests.

**Pull request merged**